### PR TITLE
Refactor: Create setCurrentUserToNavBar

### PR DESCRIPTION
**Pull request by maestromac**
## What type of PR is this? (check all applicable)
- [x] Refactor
- [x] Bug Fix

## Description
There were functionally duplicate codes, so I removed one of them and extracted it into a helper method.

## Related Tickets & Documents
n/a

## QA Instructions, Screenshots, Recordings
You should be able to browse the site, and see your username and profile image normally on the nav bar.

## Added tests?
- [x] no, because they aren't needed

## Added to documentation?
- [x] no documentation needed

## [optional] Are there any post deployment tasks we need to perform?
n/a

### CODE REVIEW

File: app/assets/javascripts/initializers/initializeBaseUserData.js [hunk 1]
```diff
@@ -103,28 +103,18 @@ function addRelevantButtonsToComments(user) {
...
-    '<span class="fw-medium block">' +
-    user.name +
-    '</span>' +
-    '<small class="fs-s color-base-50">@' +
-    user.username +
-    '</small>' +
-    '</div></a>';
-  document.getElementById(
-    'user-profile-link-placeholder',
-  ).innerHTML = userProfileLinkHTML;
```
Comment (Zhao-Andy) [inline app/assets/javascripts/initializers/initializeBaseUserData.js:121]: Is this `.innerHTML = userProfileLinkHTML` not needed anymore?

Comment (maestromac) [reply app/assets/javascripts/initializers/initializeBaseUserData.js:121]: Line 108-111 is doing the same thing, so this was a duplicate functionallity.

Comment (Zhao-Andy) [reply app/assets/javascripts/initializers/initializeBaseUserData.js:121]: Ohh I see 👍

**Review by Zhao-Andy — APPROVED** [review]
LGTM!

**Review by benhalpern — APPROVED** [review]

**Pull request merged**